File: 1-two-sum/golang/solution.go
package golang

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for k, v := range nums {
		if index, ok := m[target-v]; ok {
			return []int{k, index}
		}
		m[v] = k
	}
	return []int{-1, -1}
}

File: 1-two-sum/python/solution.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        dct = {}
        for i, el in enumerate(nums):
            if target - el in dct:
                return i, dct[target - el]
            dct[el] = i
        return -1, -1